ksqldb-engine/src/main/java/io/confluent/ksql/engine/EngineExecutor.java

```diff
@@ -221,6 +314,41 @@ private ExecutorPlans planQuery(
     return new ExecutorPlans(logicalPlan, physicalPlan);
   }
 
+  private LogicalPlanNode buildAndValidateLogicalPlan(
+      final ConfiguredStatement<?> statement,
+      final ImmutableAnalysis analysis,
+      final KsqlConfig config
+  ) {
+    final OutputNode outputNode = new LogicalPlanner(config, analysis, engineContext.getMetaStore())
```

Comment:
So we're just using the existing, unchanged `LogicalPlanner` to begin with?  Looking through it, it probably handles the basic logical nodes we would want.  I can see us doing more of our analysis and error checking there (e.g. not more than one key comparison for pull queries, etc. rather than doing it in the physical planner since that seems too far downstream).   If we handle more complex operations in the future, we'd obviously want to have a logical component.

Reply:
Yes. The logical plan nodes are there but they all have AST expressions and miss the analysis that is done for instance in the `PhysicalPlanBuilder`. A lot of that code in there should be moved to the logical plan